typing/typedecl.ml

```diff
@@ -1714,6 +1810,25 @@ let report_error ppf = function
         "cannot be checked"
   | Val_in_structure ->
       fprintf ppf "Value declarations are only allowed in signatures"
+  | Invalid_native_repr_attribute_payload kind ->
+      fprintf ppf "Payload of [@%s ] attribute must be \
+                   either a type either nothing"
+        (string_of_native_repr_kind kind)
+  | Multiple_native_repr_attributes ->
+      fprintf ppf "Too many [@unboxed ]/[@untagged ] attributes"
+  | Cannot_unbox_or_untag_type Unboxed ->
+      fprintf ppf "Don't know how to unbox this type. Only float, int32, \
+                   int64 and nativeint can be unboxed"
+  | Cannot_unbox_or_untag_type Untagged ->
+      fprintf ppf "Don't know how to untag this type. Only int \
+                   can be untagged"
+  | Type_does_not_match_native_repr_attribute_payload kind ->
+      fprintf ppf "Type and payload of [@%s: ] have different %s \
```

Comment:
I'm fine with the spaces, I didn't know it was standard.

I now understand the error message better, but it's not clear to me that referring explicitly to the "payload" (which is a language used to describe the notation used, not its meaning) is helpful here. I know this kind of comment is more helpful when an alternative is proposed, so here is an attempt:

    "Type %s cannot be %s as a %s" core_type method payload_type
    "Types %s and %s may not have the same %s representation." core_type payload_type method

In fact the code is manipulating two different kind of objects, "types" and "native representations", and it's unclear to me whether the payload argument is supposed to be any type (like, will using `[@untagged: size]` when `size` is an alias of `int` be advertised, or is supporting this only a choice of the current implementation?) or one of those predefined names that also denote native representations. If the payload is understood as any type, the second proposed message is better, otherwise the first is ok as well.